server/src/test/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequestTests.java

```diff
@@ -94,4 +102,104 @@ public void testPutMappingRequestSerialization() throws IOException {
             }
         }
     }
+
+    public void testToXContent() throws IOException {
+        PutMappingRequest request = new PutMappingRequest("foo");
+
+        request.type("my_type");
+
+        XContentBuilder mapping = JsonXContent.contentBuilder().startObject();
+        mapping.startObject("properties");
+        mapping.startObject("email");
+        mapping.field("type", "text");
+        mapping.endObject();
+        mapping.endObject();
+        mapping.endObject();
+        request.source(mapping);
```

Comment:
I had left a comment on this, do we need to test also the case where the source is not set hence it's null? We do cover it in `PutMappingRequest#toXContent`

Reply:
Oh, that's the new `testToXContentWithEmptySource` below.

(Sorry, all comments got hidden because the merge with master moved all these to `server/...`)